xclim/indices/_multivariate.py

```diff
@@ -1477,3 +1662,24 @@ def blowing_snow(
     out = cond.resample(time=freq).sum(dim="time")
     out.attrs["units"] = to_agg_units(out, snd, "count")
     return out
+
+
+def _check_common_sampling_freq(tas, pr):
+    tas_freq = infer_sampling_units(tas)
+    pr_freq = infer_sampling_units(pr)
+    if tas_freq != pr_freq:
+        raise ValidationError(
+            f"The sampling frequencies of the two datasets are different, tas frequency is {tas_freq} whereas pr frequency is {pr_freq}"
+        )
```

Comment:
I would rewrite this as a decorator for indices. This could serve any multivariate indicator, regardless of variables.

Reply:
It could be part of datachecks.py no ? I fell like it is the same kind of checking that is done here.
But the checks currently in datachecks are not decorators themself.

Reply:
I feel these check do not belong in `indices` : indices do not handle anything outside units and computations. This is a choice we made to limit the number of small adjustments and checks like this one in the indices code, making them easier to implement and read.  The down side to this is that when moving those checks to the _indicators_, they need to be more general, less specific to each function and case.

Here, the 4 indices were added to xclim.indicator.icclim and xclim.indicators.atmos. In both case, the base indicator class used is based on `xclim.core.indicator.Daily`. So the check you wanted is already performed (and its even stricter). 

I would simply remove it from here, a mismatch will be the user's problem.
If we really want to keep them, then I guess a decorator declared in datachecks.py would be the most elegant manner.